### PR TITLE
Tidy CLI parsing and zero config

### PR DESCRIPTION
**Pull request by wilbertcs**
Cleaned up argument parsing for zero config user scripts and testing.

### CODE REVIEW

File: source/fab/cli.py [hunk 1]
```diff
@@ -12,28 +12,25 @@
...
 from fab.steps.find_source_files import find_source_files
 from fab.steps.grab.folder import grab_folder
 from fab.steps.preprocess import preprocess_c, preprocess_fortran
+from fab.util import common_arg_parser
 
 
-def _generic_build_config(folder: Path, kwargs: Optional[Dict] = None) -> BuildConfig:
-    folder = folder.resolve()
-    kwargs = kwargs or {}
+def _generic_build_config(folder: str = '.', kwargs=None) -> BuildConfig:
```
Comment (MatthewHambley) [inline source/fab/cli.py:25]: What was the reason for falling back to path-as-string? If we can keep them as Path objects we should. The default constructor for Path gives the currently selected directory (e.g. `Path()`) if that was the problem.

File: source/fab/cli.py [hunk 2]
```diff
@@ -72,15 +69,24 @@ def calc_linker_flags():
...
+        parameters  ( Testing Only )
 
     """
-    # todo: use common_arg_parser()?
-    arg_parser = ArgumentParser()
-    arg_parser.add_argument('folder', nargs='?', default='.', type=Path)
-    arg_parser.add_argument('--version', action='version', version=f'%(prog)s {fab.__version__}')
-    args = arg_parser.parse_args()
+    kwargs = kwargs or {}
+    if Path(sys.argv[0]).parts[-1] == 'fab':
```
Comment (MatthewHambley) [inline source/fab/cli.py:85]: This seems to be checking that the executable is called as "fab". Is this correct and under what circumstances would it not be called that?

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
Just a few questions really.

File: source/fab/cli.py (hunk 2 above)
Comment (wilbertcs) [reply source/fab/cli.py:85]: 'fab' could , and is called by other things like 'pytest' so therefore the cli arguments may not apply to 'fab' which causes the argparser to fail with an invalid argument message.

File: source/fab/cli.py (hunk 1 above)
Comment (wilbertcs) [reply source/fab/cli.py:25]: This is only used to create the 'project_label' when using 'zero config' fab, so it simplifies the process.

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
That seems reasonable, please make the change you propose and add a comment to explain the name check.

File: source/fab/cli.py (hunk 1 above)
Comment (wilbertcs) [reply source/fab/cli.py]: My mistake it should have been a Path object

**Review by MatthewHambley — APPROVED** [review]
Looks like it's ready to go now.

**Pull request merged**